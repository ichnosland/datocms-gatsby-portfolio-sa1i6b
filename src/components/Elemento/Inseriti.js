/**
*
* Inseriti
*
*/

import styled, { css } from 'styled-components';
import { transparentize } from 'polished';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const Inseriti = styled.div`
  min-height: 34px;
  padding: 10px;
  margin: 20px auto;
  background-color: ${colore.ui.contrast};
  border-radius: 3px;
  ${media.desktop`
    min-height: 41px;
    border: 1px solid ${transparentize(0.5, colore.ui.darkBg)};
    ${({ giusto }) => giusto && css`
      border: 2px solid ${colore.actions.okay};
    `};
    ${({ sbagliato }) => sbagliato && css`
      border: 2px solid ${colore.actions.error};
    `};
  `};
  ${({ giusto }) => giusto && css`
    border: 2px solid ${colore.actions.okay};
  `};
  ${({ sbagliato }) => sbagliato && css`
    border: 2px solid ${colore.actions.error};
  `};
`;
