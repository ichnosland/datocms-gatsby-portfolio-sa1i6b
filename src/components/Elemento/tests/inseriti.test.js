import React from 'react';
import { shallow } from 'enzyme';

import { Inseriti } from '../Inseriti';

describe('<Inseriti />', () => {
  it('should render a <div> tag', () => {
    const renderedComponent = shallow(<Inseriti />);
    expect(renderedComponent.type()).toEqual('div');
  });

  it('should render its css w/ props giusto', () => {
    const renderedComponent = shallow(<Inseriti giusto />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props sbagliato', () => {
    const renderedComponent = shallow(<Inseriti sbagliato />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
