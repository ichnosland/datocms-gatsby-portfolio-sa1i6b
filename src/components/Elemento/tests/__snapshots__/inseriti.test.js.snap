// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inseriti /> should render its css w/ props giusto 1`] = `
.c0 {
  min-height: 34px;
  padding: 10px;
  margin: 20px auto;
  background-color: rgb(255,255,255);
  border-radius: 3px;
  border: 2px solid rgb(153,226,17);
}

@media (min-width:64.0625em) {
  .c0 {
    min-height: 41px;
    border: 1px solid rgba(209,209,209,0.5);
    border: 2px solid rgb(153,226,17);
  }
}

<div
  className="c0"
/>
`;

exports[`<Inseriti /> should render its css w/ props sbagliato 1`] = `
.c0 {
  min-height: 34px;
  padding: 10px;
  margin: 20px auto;
  background-color: rgb(255,255,255);
  border-radius: 3px;
  border: 2px solid rgb(255,0,0);
}

@media (min-width:64.0625em) {
  .c0 {
    min-height: 41px;
    border: 1px solid rgba(209,209,209,0.5);
    border: 2px solid rgb(255,0,0);
  }
}

<div
  className="c0"
/>
`;
